frontend/apps/reader/modules/readerzooming.lua

```diff
@@ -345,54 +490,109 @@ function ReaderZooming:genSetZoomModeCallBack(mode)
 end
 
 function ReaderZooming:setZoomMode(mode, no_warning)
-    if not no_warning and self.ui.view.page_scroll and self.paged_modes[mode] then
-        UIManager:show(InfoMessage:new{
-            text = T(_([[
+    if not no_warning and self.ui.view.page_scroll then
+        local message
+        if self.paged_modes[mode] then
+            message = T(_([[
 %1
 
-In combination with continuous view (scroll mode), this can cause unexpected vertical shifts when turning pages.]]), self.paged_modes[mode]),
-            timeout = 5,
-        })
+In combination with continuous view (scroll mode), this can cause unexpected vertical shifts when turning pages.]]),
+                self.paged_modes[mode])
```

Comment:
Indent is off here, it's aligned with the `if` instead of the `T()`. (Might still be off in suggestion; I simply added 4.)
```suggestion
                    self.paged_modes[mode])
```